### DISCUSSION TITLE
Filtering with a related table does not work

### DISCUSSION THREAD

**Discussion by kolaente** [filamentphp/filament#6571] [Help] [answered]
I have an `items` model with a relation to a `box` via a `box_item` table. I'm trying to add a filter to show all items which have a relation. 

I tried using a subquery:

```php
Filter::make('in_box')
    ->query(fn(Builder $query): Builder => $query->where(function (Builder $builder) {
        $builder->select('*')
            ->from('box_item')
            ->where('box_item.count', '>', 0)1;
    })),
```

Or a direct join with a where clause:

```php
Filter::make('in_box')
    ->query(fn(Builder $query): Builder => $query
        ->leftJoin('box_item', 'item_id', '=', 'items.id')
        ->where('box_item.count', '>', 0)
    ),
```

Using the filter results in this error message, no matter if I try to join directly or use a subquery:

![image](https://github.com/filamentphp/filament/assets/13721712/fe7a76d0-02da-4a70-80d5-96212b687204)

Error in Flare: https://flareapp.io/share/omwkay9m

It looks like Filament ignores all `join` clauses.

Is this a bug or am I using it wrong?

**Comment by danharrin** [comment, marked as answer]
> Is this a bug or am I using it wrong?

Using it wrong, you don't need an explicit join for this. Make sure you have a relationship defined

```php
->query(fn(Builder $query): Builder => $query->whereRelation('boxItem', 'count', '>', 0))
```

**Reply by kolaente** [reply]
Awesome, thank you!
